### PR TITLE
Feature/some improvements to error handling

### PR DESCRIPTION
**Pull request by jcalonso**
- Added logging for 500 errors
- Use base layout for generic errors
- Handle json / web separately

### CODE REVIEW

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -34,34 +35,36 @@ public function report(Exception $e)
...
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  \Exception  $e
+     * @param  \Illuminate\Http\Request $request
+     * @param  \Exception $e
      * @return \Illuminate\Http\Response
      */
     public function render($request, Exception $e)
     {
-        if (env('APP_DEBUG') != true) {
```
Comment (cydrobolt) [inline app/Exceptions/Handler.php]: Why was this removed?

Comment (cydrobolt) [inline app/Exceptions/Handler.php]: We shouldn't show these error pages if APP_DEBUG is true.

Comment (jcalonso) [inline app/Exceptions/Handler.php]: Its changing the behaviour of the application based on an environment variable.
If you cannot be confident of what the application does while its on dev / prod  or with debug mode, then it reduces your confidence while doing changes or trying to replicate bugs locally. Instead I added logging for the unhappy paths.

If you really want to preserve this functionality let me know and I will revert that change.

😁

Comment (cydrobolt) [inline app/Exceptions/Handler.php]: Ah, I see. Removing the APP_DEBUG clause in the Handler actually mutes all Laravel errors, not just errors occurring in the path. It's much easier to fix bugs and work on new features when you can see Laravel's debug page.

Comment (jcalonso) [inline app/Exceptions/Handler.php]: Ok, reverted 👍

**Comment by jcalonso** [comment]
Happy to merge @cydrobolt  ?

File: resources/views/errors/generic.blade.php [hunk 2]
```diff
@@ -1 +1,7 @@
-{{ $status_code }} {{ $status_message }}
+@extends('layouts.base')
+
+@section('content')
+    <h1>{{ $status_code }}</h1>
```
Comment (cydrobolt) [inline resources/views/errors/generic.blade.php]: There's a reason we didn't use the `layouts.base` for this generic error page. I remember it had something to do with API requests also hitting this "stylized" error page. Until that problem is fixed, this modification won't help API consumers.

Comment (jcalonso) [inline resources/views/errors/generic.blade.php]: In that case I will add logic to the Exception Handler to respond accordingly depending on the request type (api or web)

Comment (jcalonso) [inline resources/views/errors/generic.blade.php]: @cydrobolt Made the changes, now we handle json request separate from web requests.
Let me know if there are any other concerns you have to merge this pull request.

**Comment by cydrobolt** [comment]
A new handler for API requests and proper JSON encoding for JSON API requests was implemented in #307. #307 should have addressed most of the problems raised in this issue.

I've opened #327 to address internal API endpoints.
